plugins/metalink/utils.c

```diff
@@ -467,28 +466,30 @@ TDNFXmlParseData(
         BAIL_ON_TDNF_ERROR(dwError);
     }
 
+    if (elementInfo->startElement == NULL)
+        return;
+
     if(!strcmp(elementInfo->startElement, TAG_NAME_FILE))
     {
         elementInfo->dwError = TDNFParseFileTag(userData);
         BAIL_ON_TDNF_ERROR(elementInfo->dwError);
     }
     else if(!strcmp(elementInfo->startElement, TAG_NAME_SIZE))
     {
-        //Get File Size.
-        TDNFAllocateStringN(val, len, &size);
+        char size_buf[12], *p = size_buf;
+        const char *q = val;
 
-        if(!size)
+        while (*q && q < val + len && p < size_buf + sizeof(size_buf) && isdigit(*q))
```

Comment:
This should actually be `p < size_buf + sizeof(size_buf) - 1`. I'll fix it later.

Reply:
Done.